cmd/root.go

```diff
@@ -268,31 +269,44 @@ func pullAndRebase(container string) (imgid string, changed bool) {
 		authArgs = append(authArgs, "--authfile", kubeletAuthFile)
 	}
 
-	// Use skopeo to canonicalize to $name@$digest, so we can refer to it reliably
-	skopeoArgs := []string{"inspect"}
-	skopeoArgs = append(skopeoArgs, authArgs...)
-	skopeoArgs = append(skopeoArgs, fmt.Sprintf("docker://%s", container))
-	output := utils.RunExt(true, numRetriesNetCommands, "skopeo", skopeoArgs...)
-
-	var imagedata types.ImageInspection
-	json.Unmarshal([]byte(output), &imagedata)
-	imgid = fmt.Sprintf("%s@%s", imagedata.Name, imagedata.Digest)
-	glog.Infof("Resolved to: %s", imgid)
+	// If we're passed a non-canonical image, resolve it to its sha256 now
+	isCanonicalForm := true
+	if _, err := getRefDigest(container); err != nil {
+		isCanonicalForm = false
+		// In non-canonical form, we pull unconditionally right now
+		args := []string{"pull", "-q"}
+		args = append(args, authArgs...)
+		args = append(args, container)
+		utils.RunExt(false, numRetriesNetCommands, "podman", args...)
+	} else {
+		targetMatched, err := compareOSImageURL(previousPivot, container)
```

Comment:
Hmm, I see the reasoning in the commit message, though it wouldn't be hard to just adjust the logic so we still do this in the developer case too right? I.e.

```
if !canonical {
  podman pull
  podman inspect
  imgid = ...
} else {
  imgid = container
}
```

?

Then later on, `podman pull` by imgid, so in the developer case it's just a no-op.

Reply:
IOW, I'm not sure if

> Basically this PR ends up in a place where the non-canonical
path only loses out on the optimization of avoiding pulling the
image in the case where we already have that target.

is true since the `targetMatched` logic isn't done unconditionally, right? I.e. we can hit the `Old and new refs are equal` error.

Reply:
> i.e. we can hit the Old and new refs are equal error.

That's [not an error anymore though](https://github.com/projectatomic/rpm-ostree/pull/1807)